src/all/ninemanga/src/eu/kanade/tachiyomi/extension/all/ninemanga/NineMangaFactory.kt

```diff
@@ -59,10 +60,35 @@ class NineMangaEs : NineManga("NineMangaEs", "https://es.ninemanga.com", "es") {
     }
 
     private val imgRegex = Regex("""all_imgs_url\s*:\s*\[\s*([^]]*)\s*,\s*]""")
+    private val redirectRegex = Regex("""window\.location\.href\s*=\s*["'](.*?)["']""")
 
     override fun pageListParse(document: Document): List<Page> {
+        val redirectScript = document.selectFirst("body > script:containsData(window.location.href)")?.data()
+
+        if (redirectScript != null) {
+            var redirectUrl = redirectRegex.find(redirectScript)?.groupValues?.get(1)
+                ?: return super.pageListParse(document)
+
+            val documentLocation = document.location()
+
+            if (!redirectUrl.startsWith("http")) {
+                val documentDomain = documentLocation.substringAfter("://").substringBefore("/")
+                redirectUrl = "https://$documentDomain/${redirectUrl.removePrefix("/")}"
+            }
```

Comment:
Use `OkHttp` for this. Is `redirectUrl` always `/`...something or `http`...something? Then you can use:

```suggestion
            if (!redirectUrl.startsWith("http")) {
                redirectUrl = document.location().toHttpUrl().newBuilder()
                    .encodedPath(redirectUrl)
                    .build()
                    .toString()
            }
```